python/openassetio/hostApi/ManagerFactory.py

```diff
@@ -0,0 +1,61 @@
+#
+#   Copyright 2022 The Foundry Visionmongers Ltd
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+"""
+@namespace openassetio.hostApi.Manager
+A single-class module, providing the ManagerFactory class.
+
+@todo This module is a temporary measure until we reach parity between
+C++ and Python Manager and HostSession classes.
+"""
+
+from .. import _openassetio  # pylint: disable=no-name-in-module
+
+from .Manager import Manager
+from ..managerApi.HostSession import HostSession
+
+
+__all__ = ['ManagerFactory']
+
+
+class ManagerFactory(_openassetio.hostApi.ManagerFactory):
```

Comment:
[question] The need for this class is rather unfortunate. We're missing functionality in the C++ bound `HostSession` and `Manager` classes, so must override the `create*` methods to use the Python classes instead.

An alternative possiblity, which would keep the Python sources cleaner, may be to override `createManager` and `createManagerForInterface` in pybind, using `py::module::import()` to get the Python `Manager` (and `HostSession`) class and instantiate that instead.  

However for `createMananger` that would entail giving pybind access to `ManagerFactory`'s private members to gather the dependencies required for the `Manager` constructor.  There are probably other gotchas I haven't thought of too.

Reply:
I think the solution you have is eloquent and spot on for now, its also very easy to remove.